code/clue.js

```diff
@@ -1,59 +1,396 @@
+  name: 'Study',
+  image: 'assets/study.jpg'
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const library = {
+  name: 'Library',
+  image: 'assets/library.jpg'
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const billiardRoom = {
+  name: 'Billiard Room',
+  image: 'assets/billiard.jpg'
+};
 
+const lounge = {
+  name: 'Lounge',
+  image: 'assets/lounge.jpg'
+};
+
+const ballRoom = {
+  name: 'Ball Room',
+  image: 'assets/ballroom.jpg'
+};
+
+const hall = {
+  name: 'Hall',
+  image: 'assets/hall.jpg'
+};
+
+const spa = {
+  name: 'Spa',
+  image: 'assets/spa.jpg'
+};
+
+const livingRoom = {
+  name: 'Living Room',
+  image: 'assets/livingroom.jpg'
+};
+
+const observatory = {
+  name: 'Observatory',
+  image: 'assets/observatory.jpg'
+};
+
+const theater = {
+  name: 'Theater',
+  image: 'assets/theater.jpg'
+};
+
+const guestHouse = {
+  name: 'Guest House',
+  image: 'assets/guesthouse.jpg'
+};
+
+const patio = {
+  name: 'Patio',
+  image: 'assets/patio.jpg'
+};
+
+// STEP 2. NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candleStick,
+  dumbBell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  diningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballRoom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+];
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+const randomSelector = array => { //I don't understand this
+  return array[Math.floor(Math.random() * array.length)] //What does array.length do here?
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY. With a killer, a weapon and a room. The values will be set later.
+const mystery = {
+  killer: "",
+  weapon: "",
+  room: "",
+};
+
+//Function to shuffle the favourite weapons of the suspects in a randomized way.
+const shuffleWeapons = () => {
+  suspects.forEach(suspect => {
+    suspect.favorite = randomSelector(weapons).name
+  })
+}
+
+//Function to increase probability of favourite weapon to be assigned to killer:
+//THIS FUNCTION DOESN'T WORK???
+const increaseChanceOfFavorite = () => {
+  const favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favorite);
+  const favWeaponArray = new Array(7).fill(favWeapon);
+  console.log(favWeaponArray)
+  const calculatedWeapon = [...weapons, ...favWeaponArray];
+  return calculatedWeapon;
+};
```

Comment:
This code looks like it should work, but it requires that the killer is picked **before** it is executed.  
Otherwise, `mystery.killer` is `""`  because that is the default value.

An appropriate place to call this function might be after line 311, due to this.

Reply:
Fixed the increaseChanceOfFavorite-function now so it works. I call this function in the pickWeapon-function.